src/vcpkg/vcpkgpaths.cpp

```diff
@@ -993,6 +999,19 @@ namespace vcpkg
         return flatten_out(cmd_execute_and_capture_output(showcmd), Tools::GIT);
     }
 
+    bool VcpkgPaths::git_is_shallow_clone(const Path& dot_git_dir) const
+    {
+        auto cmd =
+            git_cmd_builder(dot_git_dir, dot_git_dir).string_arg("rev-parse").string_arg("--is-shallow-repository");
+        auto output = flatten_out(cmd_execute_and_capture_output(cmd), Tools::GIT);
+        if (!output)
+        {
+            Debug::println(output.error());
+            return false;
+        }
+        return Strings::case_insensitive_ascii_equals("true", Strings::trim(output.value_or_exit(VCPKG_LINE_INFO)));
```

Comment:
```suggestion
        return Strings::case_insensitive_ascii_equals("true", Strings::trim(*output.get()));
```

It will never exit here because errors were caught by the if statement above.